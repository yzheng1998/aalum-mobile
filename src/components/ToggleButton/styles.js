import styled from 'styled-components'
import theme from '../../../theme'

export const Button = styled.TouchableOpacity`
  border: 3px solid
    ${props =>
      props.isSelected ? theme.colors.gradientLeft : theme.colors.grey};
  border-radius: 15px;
  margin-right: 5px;
  margin-left: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
`

export const ButtonText = styled.Text`
  font-family: ${theme.fonts.title};
  color: ${props =>
    props.isSelected ? theme.colors.gradientLeft : theme.colors.grey};
  font-size: 18px;
  align-self: center;
  margin-left: 20px;
  margin-right: 20px;
  margin-top: 11px;
  margin-bottom: 11px;
`
