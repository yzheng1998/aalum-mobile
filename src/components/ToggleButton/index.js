import React, { Component } from 'react'
import { Button, ButtonText } from './styles'

export default class ToggleButton extends Component {
  render() {
    const { title, onPress, isSelected, disabled } = this.props
    return (
      <Button
        disabled={disabled}
        onPress={() => onPress(title)}
        isSelected={isSelected}
      >
        <ButtonText isSelected={isSelected}>{title}</ButtonText>
      </Button>
    )
  }
}
