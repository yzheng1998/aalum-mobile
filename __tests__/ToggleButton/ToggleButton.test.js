import React from 'react'
import ToggleButton from '../../src/components/ToggleButton'
import { shallow } from 'enzyme'
import mountWithTheme from '../helpers/theme'

describe('ToggleButton', () => {
  test('renders correctly', () => {
    const wrapper = mountWithTheme(<ToggleButton />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ title: 'title', isSelected: true, disabled: true })
    expect(wrapper).toMatchSnapshot()
  })

  test('handles addOrRemoveValue correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(<ToggleButton addOrRemoveValue={onPressMock} />)
    const button = wrapper.find('Button')
    button.simulate('press')

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })

  test('handles disabled correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(
      <ToggleButton addOrRemoveValue={onPressMock} disabled />
    )
    const button = wrapper.find('Button')

    expect(button.props().disabled).toBe(true)

    if (!button.props().disabled) {
      button.simulate('press')
    }
    expect(onPressMock).toHaveBeenCalledTimes(0)
  })
})
