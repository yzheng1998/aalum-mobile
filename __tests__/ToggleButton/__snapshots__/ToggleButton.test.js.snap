// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <ToggleButton>
    <Button
      onPress={[Function]}
    >
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Button",
            "getDefaultProps": [Function],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  border: 3px solid
    ",
                [Function],
                ";
  border-radius: 15px;
  margin-right: 5px;
  margin-left: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        onPress={[Function]}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderColor": "#A8A8A8",
                "borderRadius": 15,
                "borderStyle": "solid",
                "borderWidth": 3,
                "marginBottom": 5,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
              },
            ]
          }
        >
          <AnimatedComponent
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Array [
                  Object {
                    "borderColor": "#A8A8A8",
                    "borderRadius": 15,
                    "borderStyle": "solid",
                    "borderWidth": 3,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                  },
                ],
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "#A8A8A8",
                  "borderRadius": 15,
                  "borderStyle": "solid",
                  "borderWidth": 3,
                  "marginBottom": 5,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "#A8A8A8",
                    "borderRadius": 15,
                    "borderStyle": "solid",
                    "borderWidth": 3,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                    "opacity": 1,
                  }
                }
              >
                <Styled(Text)>
                  <StyledNativeComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "displayName": "Styled(Text)",
                        "inlineStyle": InlineStyle {
                          "rules": Array [
                            "
  font-family: ",
                            "Avenir-Heavy",
                            ";
  color: ",
                            [Function],
                            ";
  font-size: 18px;
  align-self: center;
  margin-left: 20px;
  margin-right: 20px;
  margin-top: 11px;
  margin-bottom: 11px;
",
                          ],
                        },
                        "propTypes": Object {
                          "accessible": [Function],
                          "adjustsFontSizeToFit": [Function],
                          "allowFontScaling": [Function],
                          "disabled": [Function],
                          "ellipsizeMode": [Function],
                          "minimumFontScale": [Function],
                          "nativeID": [Function],
                          "numberOfLines": [Function],
                          "onLayout": [Function],
                          "onLongPress": [Function],
                          "onPress": [Function],
                          "pressRetentionOffset": [Function],
                          "selectable": [Function],
                          "selectionColor": [Function],
                          "style": [Function],
                          "suppressHighlighting": [Function],
                          "testID": [Function],
                          "textBreakStrategy": [Function],
                        },
                        "render": [Function],
                        "styledComponentId": "StyledNativeComponent",
                        "target": [Function],
                        "withComponent": [Function],
                        Symbol(Symbol.hasInstance): [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#A8A8A8",
                            "fontFamily": "Avenir-Heavy",
                            "fontSize": 18,
                            "marginBottom": 11,
                            "marginLeft": 20,
                            "marginRight": 20,
                            "marginTop": 11,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#A8A8A8",
                              "fontFamily": "Avenir-Heavy",
                              "fontSize": 18,
                              "marginBottom": 11,
                              "marginLeft": 20,
                              "marginRight": 20,
                              "marginTop": 11,
                            },
                          ]
                        }
                      />
                    </Text>
                  </StyledNativeComponent>
                </Styled(Text)>
              </View>
            </View>
          </AnimatedComponent>
        </TouchableOpacity>
      </StyledNativeComponent>
    </Button>
  </ToggleButton>
</ThemeProvider>
`;

exports[`ToggleButton renders correctly 2`] = `
<ThemeProvider
  disabled={true}
  isSelected={true}
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
  title="title"
>
  <ToggleButton>
    <Button
      onPress={[Function]}
    >
      <StyledNativeComponent
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Button",
            "getDefaultProps": [Function],
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  border: 3px solid
    ",
                [Function],
                ";
  border-radius: 15px;
  margin-right: 5px;
  margin-left: 5px;
  margin-top: 5px;
  margin-bottom: 5px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        onPress={[Function]}
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderColor": "#A8A8A8",
                "borderRadius": 15,
                "borderStyle": "solid",
                "borderWidth": 3,
                "marginBottom": 5,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
              },
            ]
          }
        >
          <AnimatedComponent
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Array [
                  Object {
                    "borderColor": "#A8A8A8",
                    "borderRadius": 15,
                    "borderStyle": "solid",
                    "borderWidth": 3,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                  },
                ],
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "borderColor": "#A8A8A8",
                  "borderRadius": 15,
                  "borderStyle": "solid",
                  "borderWidth": 3,
                  "marginBottom": 5,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "marginTop": 5,
                  "opacity": 1,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "borderColor": "#A8A8A8",
                    "borderRadius": 15,
                    "borderStyle": "solid",
                    "borderWidth": 3,
                    "marginBottom": 5,
                    "marginLeft": 5,
                    "marginRight": 5,
                    "marginTop": 5,
                    "opacity": 1,
                  }
                }
              >
                <Styled(Text)>
                  <StyledNativeComponent
                    forwardedClass={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "displayName": "Styled(Text)",
                        "inlineStyle": InlineStyle {
                          "rules": Array [
                            "
  font-family: ",
                            "Avenir-Heavy",
                            ";
  color: ",
                            [Function],
                            ";
  font-size: 18px;
  align-self: center;
  margin-left: 20px;
  margin-right: 20px;
  margin-top: 11px;
  margin-bottom: 11px;
",
                          ],
                        },
                        "propTypes": Object {
                          "accessible": [Function],
                          "adjustsFontSizeToFit": [Function],
                          "allowFontScaling": [Function],
                          "disabled": [Function],
                          "ellipsizeMode": [Function],
                          "minimumFontScale": [Function],
                          "nativeID": [Function],
                          "numberOfLines": [Function],
                          "onLayout": [Function],
                          "onLongPress": [Function],
                          "onPress": [Function],
                          "pressRetentionOffset": [Function],
                          "selectable": [Function],
                          "selectionColor": [Function],
                          "style": [Function],
                          "suppressHighlighting": [Function],
                          "testID": [Function],
                          "textBreakStrategy": [Function],
                        },
                        "render": [Function],
                        "styledComponentId": "StyledNativeComponent",
                        "target": [Function],
                        "withComponent": [Function],
                        Symbol(Symbol.hasInstance): [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#A8A8A8",
                            "fontFamily": "Avenir-Heavy",
                            "fontSize": 18,
                            "marginBottom": 11,
                            "marginLeft": 20,
                            "marginRight": 20,
                            "marginTop": 11,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "alignSelf": "center",
                              "color": "#A8A8A8",
                              "fontFamily": "Avenir-Heavy",
                              "fontSize": 18,
                              "marginBottom": 11,
                              "marginLeft": 20,
                              "marginRight": 20,
                              "marginTop": 11,
                            },
                          ]
                        }
                      />
                    </Text>
                  </StyledNativeComponent>
                </Styled(Text)>
              </View>
            </View>
          </AnimatedComponent>
        </TouchableOpacity>
      </StyledNativeComponent>
    </Button>
  </ToggleButton>
</ThemeProvider>
`;
